examples/experimental/dagster-airlift/dagster_airlift/in_airflow/mark_as_migrating.py

```diff
@@ -25,21 +28,66 @@ def mark_as_dagster_migrating(
-        dag: DAG = obj
-        logger.debug(f"Checking dag with id `{dag.dag_id}` for migration state.")
-        migration_state_for_dag = migration_state.get_migration_dict_for_dag(dag.dag_id)
-        if migration_state_for_dag is None:
-            logger.debug(
-                f"Dag with id `{dag.dag_id} hasn't been marked with migration state. Skipping..."
-            )
-        else:
-            dag.tags.append(json.dumps({"DAGSTER_MIGRATION_STATUS": migration_state_for_dag}))
-            logger.debug(
-                f"Dag with id `{dag.dag_id}` has been marked with migration state. Adding state to tags for dag."
+    dag_vars_to_mark = set()
+    task_vars_to_migrate = set()
+    all_dags_by_id = {}
+    for var, obj in global_vars.items():
+        if isinstance(obj, DAG):
+            dag: DAG = obj
+            if migration_state.dag_is_being_migrated(obj.dag_id):
+                logger.debug(f"Dag with id `{dag.dag_id}` has migration state.")
+                dag_vars_to_mark.add(var)
+            else:
+                logger.debug(
+                    f"Dag with id `{dag.dag_id} has no associated migration state. Skipping..."
+                )
+            all_dags_by_id[obj.dag_id] = obj
+        if isinstance(obj, BaseOperator) and migration_state.get_migration_state_for_task(
+            dag_id=obj.dag_id, task_id=obj.task_id
+        ):
+            task_vars_to_migrate.add(var)
+
+    for var in dag_vars_to_mark:
+        dag: DAG = global_vars[var]
+        logger.debug(f"Tagging dag {dag.dag_id} as migrating.")
+        dag.tags.append(
+            json.dumps(
+                {"DAGSTER_MIGRATION_STATUS": migration_state.get_migration_dict_for_dag(dag.dag_id)}
             )
-            num_dags += 1
-    logger.info(f"Marked {num_dags} dags as migrating to dagster{suffix}.")
+        )
+    logging.debug(f"Marked {len(dag_vars_to_mark)} dags as migrating to dagster via tag.")
+
+    for var in task_vars_to_migrate:
+        original_op: BaseOperator = global_vars[var]
+        logger.debug(
+            f"Creating new operator for task {original_op.task_id} in dag {original_op.dag_id}"
+        )
+        new_op = build_dagster_task(
+            task_id=original_op.task_id,
+            owner=original_op.owner,
+            email=original_op.email,
+            email_on_retry=original_op.email_on_retry,
+            email_on_failure=original_op.email_on_failure,
+            retries=original_op.retries,
+            retry_delay=original_op.retry_delay,
+            retry_exponential_backoff=original_op.retry_exponential_backoff,
+            max_retry_delay=original_op.max_retry_delay,
+            start_date=original_op.start_date,
+            end_date=original_op.end_date,
+            depends_on_past=original_op.depends_on_past,
+            wait_for_downstream=original_op.wait_for_downstream,
+            params=original_op.params,
+            doc_md="This task has been migrated to dagster.",
```

Comment:
Actually might be worth being super defensive on all these accesses and explicitly testing for if the attribute exists just to be paranoid. It's python so people can do anything with monkeypatching.

Reply:
That makes sense yea. I wish there was a more generic way to do this.